src/Raven.Server/Documents/Replication/ResolveConflictOnReplicationConfigurationChange.cs

```diff
@@ -222,14 +222,52 @@ private void UpdateScriptResolvers()
 
         public void PutResolvedDocument(
            DocumentsOperationContext context,
-           DocumentConflict conflict)
+           DocumentConflict resolved,
+           DocumentConflict incoming = null)
         {
-            if (conflict.Doc == null)
+            if (incoming != null)
             {
-                using (Slice.External(context.Allocator, conflict.LowerId, out Slice lowerId))
+                // we resolved the conflict on the fly, so we save the remote documents as revisions
```

Comment:
We need to save all the conflicts that we resolved here.

Reply:
Those are 2 different code paths. The existing conflicts are saved when we put the resolved document.